unit-tests/live/syncer/test-throughput.cpp

```diff
@@ -0,0 +1,153 @@
+            return;
+        auto frame_num = f.get_frame_number();
+        auto fps = f.get_frame_metadata(RS2_FRAME_METADATA_ACTUAL_FPS);
+        if (!frames_num_info[fps].empty() && frames_num_info[fps].back() == frame_num) // check if frame is already processed
+            return;
+        frames_num_info[fps].push_back(frame_num);
+    };
+
+    auto validate_ratio = [&](unsigned int delta, configuration test)
+    {
+        for (auto f : frames_num_info)
+        {
+            std::cout << "Infrared 1 : " << f.first << " fps, " << f.second.size() << " frames" << std::endl;
+            if (test == STOP) // after changing exposure to 18000, fps should be changed 60 -> 30
+            {
+                double ratio = (double)f.first / prev_fps;
+                REQUIRE(ratio < 0.6);
+            }
+            prev_fps = (double)f.first;
+            double calc_fps = (double)f.second.size() / delta;
+            double fps_ratio = calc_fps/ f.first;
+            CAPTURE(calc_fps, f.second.size(), delta, f.first);
+            REQUIRE(fps_ratio > 0.8);
+            
+        }
+    };
+
+    for (auto& test : tests)
+    {
+        CAPTURE(test);
+        frames_num_info.clear();
+        ir_sensor.set_option(RS2_OPTION_EXPOSURE, 1);
+        ir_sensor.open(ir_stream_profile); // ir streams in all configurations
+        ir_sensor.start(sync);
+
+        if (test == IR_RGB_EXPOSURE)
+        {
+            rgb_sensor.open(rgb_stream_profile);
+            rgb_sensor.start(sync);
+        }
+
+        std::cout << "==============================================" << std::endl;
+        std::cout << "Configuration " << test << std::endl << std::endl;
+
+        auto t_start = std::chrono::system_clock::now();
+        auto t_end = std::chrono::system_clock::now();
+        unsigned int delta = (unsigned int) std::chrono::duration_cast<std::chrono::seconds>(t_end - t_start).count();
+        while (delta < RECEIVE_FRAMES_TIME)
+        {
+            auto fs = sync.wait_for_frames();
+            for (const rs2::frame& ff : fs)
+            {
+                process_frame(ff);
+            }
+            t_end = std::chrono::system_clock::now();
+            delta = (unsigned int) std::chrono::duration_cast<std::chrono::seconds>(t_end - t_start).count();
+
+            if (test == IR_RGB_EXPOSURE && delta >= RECEIVE_FRAMES_TIME) // modify exposure 5 seconds after start streaming ( according to repro description )
+            {
+                validate_ratio(delta, test);
+                test = STOP;
+                delta = 0;
+                frames_num_info.clear();
+                t_start = std::chrono::system_clock::now();
+                ir_sensor.set_option(RS2_OPTION_EXPOSURE, 18000); // set exposure value to x > 1000/fps
+                std::this_thread::sleep_for(std::chrono::milliseconds(500)); // wait 200 msec to process FW command
+            }
+        }
+        validate_ratio(delta, test);
```

Comment:
After changing the exposure the test should stream for at least 5-10 seconds to gather data for analysis.
Run the check after sensors are stopped.

Reply:
That's what I did in the test